File: backend/src/main/java/br/com/calciolari/datahub/imports/domain/parser/SourceLocator.java
package br.com.calciolari.datahub.imports.domain.parser;

/**
 * Optional locator back to the source report (page/record/offset). Populated only
 * from observed parser evidence — never invented placeholders.
 */
public record SourceLocator(
		Integer page,
		Integer recordIndex,
		Long byteOffset,
		String detail
) {
	public static SourceLocator empty() {
		return new SourceLocator(null, null, null, null);
	}
}

File: backend/src/Calciolari.DataHub/Imports/Domain/Parser/SourceLocator.cs
namespace Calciolari.DataHub.Imports.Domain.Parser;

/// <summary>
/// Optional locator back to the source report (page/record/offset). Populated only
/// from observed parser evidence — never invented placeholders.
/// </summary>
public sealed record SourceLocator(int? Page, int? RecordIndex, long? ByteOffset, string? Detail)
{
    public static SourceLocator Empty { get; } = new(null, null, null, null);

    public override string ToString() =>
        $"SourceLocator[page={Fmt(Page)}, recordIndex={Fmt(RecordIndex)}, byteOffset={Fmt(ByteOffset)}, detail={Detail ?? "null"}]";

    private static string Fmt<T>(T? value) where T : struct => value is null ? "null" : value.Value.ToString()!;
}
